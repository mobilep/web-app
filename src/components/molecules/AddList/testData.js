export const users = [
	{
		id: 1,
		fullName: 'John Smith',
		initials: 'JS',
	}, {
		id: 2,
		fullName: 'Philippe Starck',
		initials: 'PS',
	}, {
		id: 3,
		fullName: 'Suzy Salmon',
		initials: 'SS',
	},
];

export const allUsers = [
	{
		id: 4,
		fullName: 'John Rebele',
		initials: 'JR',
	}, {
		id: 5,
		fullName: 'John Klokdan',
		initials: 'JK',
	}, {
		id: 6,
		fullName: 'John Makeba',
		initials: 'JM',
	},
];
