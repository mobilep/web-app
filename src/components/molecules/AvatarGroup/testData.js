export const avatars = [
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
];

export const avatarsLess = [
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
];
