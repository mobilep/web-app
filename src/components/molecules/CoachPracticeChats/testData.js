/* eslint-disable max-len */
export const practiceChats = [
	{
		currentUserId: '5d6e509093b6e912b83a65ce',
		_id: '5dad8ef5d3c420b10c618100',
		_coach: {
			_id: '5d6e509093b6e912b83a65ce',
			firstName: 'Roman',
			lastName: 'Yavoriv',
			avatarId: '5f549515-ee86-40f2-9104-5c6a8a80beb7',
			devices: 'Array[0]',
			avatarColor: '58c9ef',
			lang: 'en',
			avatar_sm: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/100/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			avatar_md: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/640/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			avatar_lg: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/1024/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			havePassword: false,
			name: 'Roman Yavoriv',
			extraInformation: 'Array[7]',
		},
		_scenario: {
			_id: '5dad8ef546396073dd363907',
			groupChat: 'https://mobile-practice-dev.firebaseio.com/mpchat/5dad8ef546396073dd363909',
			isPracticed: true,
			isActive: true,
			name: '2nd scenario',
			video: 'Object',
		},
		_user: {
			_id: '5d70bc6d93b6e912b83a65d5',
			firstName: 'Taras',
			lastName: 'Kvas',
			avatarId: '2e3db607-9ec1-45f4-a7e5-4bed03912dd5',
			devices: 'Array[0]',
			avatarColor: 'fd524f',
			lang: 'en',
			avatar_sm: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/100/2e3db607-9ec1-45f4-a7e5-4bed03912dd5.jpg',
			avatar_md: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/640/2e3db607-9ec1-45f4-a7e5-4bed03912dd5.jpg',
			avatar_lg: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/1024/2e3db607-9ec1-45f4-a7e5-4bed03912dd5.jpg',
			havePassword: false,
			name: 'Taras Kvas',
			extraInformation: 'Array[7]',
		},
		__v: 0,
		createdAt: '2019-10-21T10:56:53.554Z',
		updatedAt: '2019-10-29T09:19:42.764Z',
		userEvaluatedAt: null,
		coachEvaluatedAt: null,
		hasBestPractice: false,
		hasUserMessage: true,
		isActive: true,
		lastMessageFrom: 'learner',
		status: 'current',
		coachMark: [],
		userMark: [],
		coachAudiosCount: 0,
		coachPhotosCount: 0,
		coachFilesCount: 0,
		coachVideosCount: 0,
		coachMessagesCount: 4,
		userAudiosCount: 0,
		userPhotosCount: 1,
		userFilesCount: 0,
		userVideosCount: 0,
		userMessagesCount: 19,
		_inbox: {
			_id: '5dad8ef546396073dd36390c',
			updatedAt: '2019-10-29T09:20:28.648Z',
			createdAt: '2019-10-21T10:56:53.544Z',
			_user: '5d6e509093b6e912b83a65ce',
			_recipient: '5d70bc6d93b6e912b83a65d5',
			__v: 0,
			inboxId: 'chats/5d6e509093b6e912b83a65ce/5dad8ef546396073dd363907/5dad8ef5d3c420b10c618100/5dad8ef546396073dd36390c',
			lastReadMessage: '-LsLo4OhHveQpLP5JXKy',
			unreadMessagesCount: 0,
			type: 'practice',
			status: 'read',
			message: 'Object',
			firstResponseAt: '2019-10-28T13:25:34.142Z',
			firstMessageAt: '2019-10-28T13:09:17.159Z',
			isActive: true,
		},
		state: 'waitingOnFeedback',
		messages: [
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
		],
	},
	{
		currentUserId: '5d6e509093b6e912b83a65ce',
		_id: '5dad8ef5d3c420b10c618102',
		_coach: {
			_id: '5d6e509093b6e912b83a65ce',
			firstName: 'Roman',
			lastName: 'Yavoriv',
			avatarId: '5f549515-ee86-40f2-9104-5c6a8a80beb7',
			devices: 'Array[0]',
			avatarColor: '58c9ef',
			lang: 'en',
			avatar_sm: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/100/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			avatar_md: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/640/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			avatar_lg: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/1024/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			havePassword: false,
			name: 'Roman Yavoriv',
			extraInformation: 'Array[7]',
		},
		_scenario: {
			_id: '5dad8ef546396073dd363907',
			groupChat: 'https://mobile-practice-dev.firebaseio.com/mpchat/5dad8ef546396073dd363909',
			isPracticed: true,
			isActive: true,
			name: '2nd scenario',
			video: 'Object',
		},
		_user: {
			_id: '5da86a3716973b29347321ce',
			firstName: 'Roman',
			lastName: 'Yavoriv',
			devices: 'Array[0]',
			avatarColor: '00cdbb',
			lang: 'en',
			avatar_sm: null,
			avatar_md: null,
			avatar_lg: null,
			havePassword: false,
			name: 'Roman Yavoriv',
			extraInformation: 'Array[7]',
		},
		__v: 0,
		createdAt: '2019-10-21T10:56:53.563Z',
		updatedAt: '2019-10-29T09:18:36.340Z',
		userEvaluatedAt: '2019-10-29T09:16:14.949Z',
		coachEvaluatedAt: null,
		hasBestPractice: false,
		hasUserMessage: false,
		isActive: true,
		lastMessageFrom: 'coach',
		coachMark: [
			'Object',
		],
		userMark: [],
		coachAudiosCount: 0,
		coachPhotosCount: 0,
		coachFilesCount: 0,
		coachVideosCount: 0,
		coachMessagesCount: 2,
		userAudiosCount: 0,
		userPhotosCount: 0,
		userFilesCount: 0,
		userVideosCount: 0,
		userMessagesCount: 0,
		_inbox: {
			_id: '5dad8ef546396073dd36390d',
			updatedAt: '2019-10-29T09:19:11.848Z',
			createdAt: '2019-10-21T10:56:53.547Z',
			_user: '5d6e509093b6e912b83a65ce',
			_recipient: '5da86a3716973b29347321ce',
			__v: 0,
			inboxId: 'chats/5d6e509093b6e912b83a65ce/5dad8ef546396073dd363907/5dad8ef5d3c420b10c618102/5dad8ef546396073dd36390d',
			lastReadMessage: '-LsLnpAfH4s1IOZsQrgY',
			unreadMessagesCount: 0,
			type: 'practice',
			status: 'read',
			message: 'Object',
			firstResponseAt: null,
			firstMessageAt: '2019-10-29T09:18:27.588Z',
			isActive: true,
		},
		state: 'completed',
		messages: [
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
		],
	},
];

export const activeChatId = practiceChats[0]._id;
