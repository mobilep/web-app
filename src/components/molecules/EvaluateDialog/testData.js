export const evaluationList = [
	{
		id: '1',
		title: 'Contactâ€™s attention grabbed',
	}, {
		id: '2',
		title: 'Self Introduction',
	}, {
		id: '3',
		title: 'Purpose of call stated',
	},
];
