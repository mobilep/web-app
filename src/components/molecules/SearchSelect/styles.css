.SearchSelect {
	position: relative;
	width: 100%;
}

.SearchSelect .SearchSelect-selectList {
	width: calc(100% - 36px);
	max-height: 400px;
	overflow-y: auto;
	overflow-x: hidden;
	text-overflow: ellipsis;
	margin-left: 36px;
	position: absolute;
	z-index: 1;
}
