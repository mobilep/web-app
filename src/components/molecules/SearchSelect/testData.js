export const items = [{
	id: '1',
	value: 'first item',
}, {
	id: '2',
	value: 'second item',
}, {
	id: '3',
	value: 'third item',
}];
