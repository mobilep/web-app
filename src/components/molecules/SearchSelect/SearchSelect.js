import React, { Component, createRef } from 'react';
import PropTypes from 'prop-types';
import { Subject } from 'rxjs';
import { tap, debounceTime, takeWhile, map } from 'rxjs/operators';
import { AddItem } from '../../atoms';
import { SelectList } from '..';
import './styles.css';

export default class SearchSelect extends Component {
	static propTypes = {
		className: PropTypes.string,
		createItemId: PropTypes.symbol,
		itemIdKey: PropTypes.string,
		itemValueKey: PropTypes.string,
		items: PropTypes.array,
		notFoundItemId: PropTypes.symbol,
		onCreateItem: PropTypes.func,
		onSearch: PropTypes.func,
		onSelect: PropTypes.func,
		placeholder: PropTypes.string,
	}

	static defaultProps = {
		onSearch: () => {},
		onSelect: () => {},
	}

	state = {
		inputValue: '',
		showList: false,
		selectedItemIndex: null,
	}

	ref = createRef();

	handleItemMouseOver = (selectedItemIndex) =>
		this.setState({ selectedItemIndex })

	componentDidMount () {
		document.addEventListener('click', this.handleDocumentClick);

		this.inputChangeStream$ = new Subject();
		this.inputChangeStream$
			.pipe(
				tap((inputValue) =>
					this.setState({
						selectedItemIndex: null,
						showList: true,
						inputValue,
					})),
				map((inputValue) => inputValue.trim()),
				debounceTime(150),
				takeWhile(() => !this.inputChangeStream$.isStopped)
			)
			.subscribe(this.props.onSearch);
	}

	componentWillUnmount () {
		document.removeEventListener('click', this.handleDocumentClick);
		this.inputChangeStream$.unsubscribe();
	}

	handleDocumentClick = (event) => {
		const { target } = event;
		const { showList } = this.state;
		if (!this.ref.current) return;
		const isSearchSelectAreaClick = this.ref.current.contains(target);
		if (!showList || isSearchSelectAreaClick) return;
		this.setState({ showList: false });
	}

	getDecreasedIndex (selectedItemIndex = this.state.selectedItemIndex) {
		let nextIndex;
		if (selectedItemIndex === null) {
			nextIndex = this.props.items.length - 1;
		} else {
			nextIndex = selectedItemIndex - 1;
		}
		return nextIndex;
	}

	getIncreasedIndex (selectedItemIndex = this.state.selectedItemIndex) {
		let nextIndex;
		if (selectedItemIndex === null) {
			nextIndex = 0;
		} else {
			nextIndex = selectedItemIndex + 1;
		}
		return nextIndex;
	}

	handleKeyDown = (event) => {
		const key = event.key.toLowerCase();
		if (!['arrowdown', 'arrowup', 'enter'].includes(key)) return;
		const { items, itemIdKey } = this.props;
		const { length: itemsLength } = items;
		const { selectedItemIndex: currentItemIndex } = this.state;
		let selectedItemIndex;
		switch (key) {
			case 'enter':
				if (this.state.selectedItemIndex !== null) {
					this.handleSelect(this.props.items[this.state.selectedItemIndex][itemIdKey]);
				}
				return;
			case 'arrowdown':
				selectedItemIndex = currentItemIndex !== itemsLength - 1
					? this.getIncreasedIndex()
					: null;
				return this.setState({ selectedItemIndex });
			case 'arrowup':
				selectedItemIndex = currentItemIndex !== 0
					? this.getDecreasedIndex()
					: null;
				return this.setState({ selectedItemIndex });
			default:
				break;
		}
	}

	handleSelect = (selectedId) => {
		const { itemIdKey, items, createItemId, notFoundItemId, onCreateItem } = this.props;
		if (selectedId === notFoundItemId) return;
		const selectedCriteria = items.find((item) => item[itemIdKey] === selectedId);
		if (selectedId === createItemId) {
			onCreateItem(selectedCriteria.name);
		} else {
			this.props.onSelect(selectedCriteria);
		}
		this.setState({ selectedItemIndex: null, showList: false, inputValue: '' });
	}

	handleFocus = () =>
		this.setState({ showList: true });

	handleChange = (value) =>
		this.inputChangeStream$.next(value);

	shouldRenderList ({ showList } = this.state, { items } = this.props) {
		return showList && !!items.length;
	}

	render () {
		const { className, placeholder, items, itemIdKey, itemValueKey } = this.props;
		const { inputValue, selectedItemIndex } = this.state;
		return (
			<div ref={this.ref} className={`SearchSelect ${className}`}>
				<AddItem
					maxLength={128}
					placeholder={placeholder}
					onFocus={this.handleFocus}
					onKeyDown={this.handleKeyDown}
					onChange={this.handleChange}
					value={inputValue}
				/>
				{
					this.shouldRenderList() &&
					<SelectList
						itemIdKey={itemIdKey}
						itemValueKey={itemValueKey}
						className='SearchSelect-selectList'
						selectedItem={selectedItemIndex}
						onSelect={this.handleSelect}
						items={items}
						onMouseOver={this.handleItemMouseOver}
					/>
				}
			</div>
		);
	}
}
