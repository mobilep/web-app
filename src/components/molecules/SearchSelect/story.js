/* eslint-disable no-alert */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { withState } from '@dump247/storybook-state';
import SearchSelect from './SearchSelect';
import { items } from './testData';

storiesOf('2. Molecules / SearchSelect', module)
	.add('SearchSelect', withState({ items: [] })(({ store }) => (
		<SearchSelect {...store.state} onSearch={(query) => {
			const itemsProps = items.filter(({ value }) =>
				value.toLowerCase().includes(query));
			store.set({ items: itemsProps });
		}} />
	)));
