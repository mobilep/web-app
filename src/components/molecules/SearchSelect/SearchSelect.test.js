import React from 'react';
import { shallow } from 'enzyme';

import SearchSelect from './SearchSelect';
import { userItems } from './testData';

describe('Molecule - SearchSelect', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<SearchSelect
				items={userItems}
				placeholder='Placeholder'
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

});
