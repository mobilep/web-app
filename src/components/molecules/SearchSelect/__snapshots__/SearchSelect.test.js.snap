// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule - SearchSelect Should render 1`] = `
<div
  className="SearchSelect undefined"
>
  <AddItem
    maxLength={128}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    placeholder="Placeholder"
    value=""
  />
</div>
`;
